translator/translate/logs/util/validate_fields.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator"
+)
+
+const (
+	logRetentionKey  = "retention_in_days"
+	logGroupKey      = "log_group_name"
+	logGroupClassKey = "log_group_class"
+)
+
+func ValidateLogGroupFields(logConfigs []interface{}, currPath string) []interface{} {
+	logConfigs = validateLogRetentionSettings(logConfigs, currPath)
+	logConfigs = validateLogGroupClassSettings(logConfigs, currPath)
+	return logConfigs
+}
+
+func validateLogRetentionSettings(logConfigs []interface{}, currPath string) []interface{} {
+	configMap := make(map[string]int)
+	for _, logConfig := range logConfigs {
+		if logConfigMap, ok := logConfig.(map[string]interface{}); ok {
```

Comment:
nit: I would prefer less nesting by checking for the negative case first:

```
if logConfigMap, ok := logConfig.(map[string]interface{}); !ok {
    continue 
}
if retention, ok := logConfigMap[logRetentionKey].(int); !ok {
    continue
}
 ...
 ```

Reply:
Can follow up in a separate PR? This was moved from a different file to simplify logic, not related to this PR